language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java

```diff
@@ -557,4 +562,29 @@ public void didSave(DidSaveTextDocumentParams params) {
             }
         });
     }
+
+    @Override
+    public CompletableFuture<SemanticTokens> semanticTokensFull(SemanticTokensParams params) {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                SemanticTokensCapabilities capabilities =
+                        this.clientCapabilities.getTextDocCapabilities().getSemanticTokens();
+                if (capabilities == null || !capabilities.getDynamicRegistration()) {
+                    if (!LSClientConfigHolder.getInstance(serverContext).getConfig().isEnableSemanticHighlighting()) {
+                        return new SemanticTokens(new ArrayList<>());
+                    }
+                }
```

Comment:
This was added to handle the clients that don't support dynamic registration capabilities. 

- If the dynamic registration is available and if the capability is unregistered via the  ClientConfigListener, clients don't get any invocation to the API. And, when the capability is registered, the configuration check will be skipped at `if (capabilities == null || !capabilities.getDynamicRegistration()) {`
- If the dynamic registration is not available, it always checks the configuration before generating semantic tokens.

Reply:
Whether the client does not have the dynamic registration capabilities or not, the config listener will receive the config update. There we should prompt the user to reload the client. Otherwise we are handling the capabilities in two locations. If the user does not reload the window, it will continue accepting the requests. On the other hand, client is not aware of the config change and it keeps sending the requests to the server unnecessarily. So, from the user's point of view, we should force the user to reload the window to avoid both client side's requests sending and server side's request accepting.

Reply:
Removed configuration check for non-dynamic registration scenario, as discussed.